iroh-net/src/hp/portmapper/upnp.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    fmt::Display,
+    net::SocketAddrV4,
+    time::{Duration, Instant},
+};
+
+use anyhow::Error;
+use igd::aio as aigd;
+
+/// Seconds we ask the router to maintain the port mapping. 0 means infinite.
+const PORT_MAPPING_LEASE_DURATION_SECONDS: u32 = 0;
+
+/// Maximum duration a UPnP search can take before timing out.
+const SEARCH_TIMEOUT: Duration = Duration::from_millis(1000);
+
+const PORT_MAPPING_DESCRIPTION: &str = "iroh-portmap";
+
+#[derive(Debug, Clone)]
+pub struct Mapping {
+    /// The internet Gateway device (router) used to create this mapping.
+    gateway: aigd::Gateway,
+    /// The external address obtained by this mapping.
+    external_addr: SocketAddrV4,
+    /// Local address used to create this mapping.
+    local_addr: SocketAddrV4,
+    /// Instant in which the mapping was first created or last renewed.
+    created_at: Instant,
+}
+
+impl Mapping {
+    pub(crate) async fn new(local_addr: SocketAddrV4) -> Result<Self, Error> {
```

Comment:
We usually import `anyhow::Result` directly and use `Result<Self>` (for example).

Reply:
done in [32d5a0c](https://github.com/n0-computer/iroh/pull/1117/commits/32d5a0c5022e76bd764d0c8d7431c71fb284e6ad)